idl.md

```diff
@@ -1,61 +1,90 @@
-## Proposed IDL
-```javascript
+# Proposed IDL
+
+```js
 partial interface ServiceWorkerRegistration {
-  readonly attribute SyncManager syncManager;
+  readonly attribute SyncManager sync;
+  readonly attribute PeriodicSyncManager periodicSync;
 };
 
 interface SyncManager {
   Promise<SyncRegistration> register(optional SyncRegistrationOptions options);
-  Promise<SyncRegistration> getRegistration(DOMString id);
+  Promise<SyncRegistration> getRegistration(DOMString tag);
   Promise<sequence<SyncRegistration>> getRegistrations();
-  Promise<SyncPermissionStatus> hasPermission();
-  readonly attribute unsigned long minAllowablePeriod;
+  Promise<SyncPermissionState> permissionState();
```

Comment:
The main thing I don't get is why we have such long method names still. You already need to do `navigator.sync`. At that point we should just be able to use `get()` and `getAll()` and such. Do we really expect those names to be used for something else? Same for `permissionState()` vs `permission()`. Why is longer better? We always try to shorten elsewhere.